src/ThemeLoader.js

```diff
@@ -0,0 +1,98 @@
+import cssVarsPonyfill from 'css-vars-ponyfill';
+
+const buildURL = (cdn = '', themeId = '') => {
+    if (cdn === '' && themeId === '') return '';
+
+    if (cdn.charAt(cdn.length - 1) !== '/') {
+        return `${cdn}/${themeId}`;
+    }
+
+    return `${cdn}${themeId}`;
+};
+
+const isInternetExplorer11 = () => {
+    const ua = window.navigator.userAgent;
+    return ua.indexOf('Trident/') > 0;
+};
+
+const DEFAULT_LOADING_TIMEOUT = 2000;
+
+class ThemeLoader {
+    /**
+     * Loads the theme and the CSS vars ponyfill if necessary
+     *
+     * @param {Object} themeConfig - Object that contains the configuration to initialize the theme
+     * @param {string} themeConfig.cdn - Server URL that hosts the theme
+     * @param {string} themeConfig.themeId - Theme unique idenfitier
+     * @param {number} themeConfig.maxTime - Max time to wait for the theme to be loaded
+     * @param {Object} themeConfig.ponyfillOptions - Set of options that can be used to configure the ponyfill. Options: https://www.npmjs.com/package/css-vars-ponyfill#options
+     */
+    static init({ cdn, themeId, maxTime = DEFAULT_LOADING_TIMEOUT, ponyfillOptions } = {}) {
+        const loadTheme = Promise.all([
+            ThemeLoader.applyTheme(cdn, themeId),
+            ThemeLoader.startCssVarsPonyfill(ponyfillOptions)
+        ]);
+
+        const timeout = new Promise((resolve, reject) =>
+            setTimeout(
+                () =>
+                    reject(
+                        new Error(
+                            `Theme "${buildURL(cdn, themeId)}" not loaded. Loading time expired`
+                        )
+                    ),
+                maxTime
+            )
+        );
+
+        return Promise.race([loadTheme, timeout]);
+    }
+
+    /**
+     * Loads the CSS Vars ponyfill in case the browser is Internet Explorer 11. It can also
+     * forces to load in modern browsers in case the user passes `onlyLegacy` property as false
+     */
+    static startCssVarsPonyfill(ponyfillOptions = { onlyLegacy: true }) {
+        return new Promise((resolve, reject) => {
+            if (!isInternetExplorer11() && ponyfillOptions.onlyLegacy === true) {
```

Comment:
The default object won't work if ponyfillOptions is defined, but without the `onlyLegacy` key. In that case `ponyfillOptions.onlyLegacy` will be undefined. Since the ponyfill's default is `true` anyway, I suggest to drop the default in the definition above. In anycase, should add `|| ponyfillOptions.onlyLegacy===undefined`